flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -524,13 +534,24 @@ class JobManager(
    * @param listenToEvents true if the sender wants to listen to job status and execution state
    *                       change notifications. false if not.
    */
-  private def submitJob(jobGraph: JobGraph, listenToEvents: Boolean): Unit = {
+  private def submitJob(
+      jobGraph: JobGraph,
+      listenToEvents: Boolean,
+      clientVersion: String)
+    : Unit = {
     if (jobGraph == null) {
       sender ! decorateMessage(
         Failure(
           new JobSubmissionException(null, "JobGraph must not be null.")
         )
       )
+    } else if(jobManagerVersionID != clientVersion){
+      sender ! decorateMessage(
+        Failure(
+          new JobSubmissionException(jobGraph.getJobID, "Client version mismatches that of Job " +
+            "Manager.")
```

Comment:
The version is drawn from the Jar file's package version. Maven automatically adds this info.